ttnn/cpp/ttnn/operations/data_movement/tilize_with_val_padding/tilize_with_val_padding.cpp

```diff
@@ -7,11 +7,78 @@
 #include "device/tilize_with_val_padding_op.hpp"
 #include "ttnn/common/constants.hpp"
 #include "ttnn/run_operation.hpp"
+#include "ttnn/operations/data_movement/common/common.hpp"
+#include "ttnn/operations/data_movement/reshape_view/reshape.hpp"
 
 using namespace tt::tt_metal;
 
 namespace ttnn::operations::data_movement {
 
+using OwnedTilizeValArgs = std::tuple<ttnn::Tensor>;
+using BaseTilizeValType = std::function<ttnn::Tensor(const ttnn::Tensor&)>;
+
+using MassagedTilizeVal = MassagedOperation<ttnn::Tensor, const ttnn::Tensor&>;
+using MassagedTilizeValParams = MassagedOperationParams<ttnn::Tensor, const ttnn::Tensor&>;
+
+uint TILE_WIDTH = 32;
+uint TILE_HEIGHT = 32;
+ttnn::Shape update_original_shape(ttnn::Shape& original) {
+    std::vector<uint32_t> update_original(original.rank());
+    if (original[original.rank() - 2] % TILE_HEIGHT != 0) {
```

Comment:
Could you please save `original.rank() - 2]` and `original.rank() - 1]` into variables and use the variables instead.